platform/ui/src/components/downloadDialog/DownloadDialog.js

```diff
@@ -0,0 +1,235 @@
+import './DownloadDialog.styl';
+
+import React, { PureComponent } from 'react';
+import { withTranslation } from '../../utils/LanguageProvider';
+import Modal from 'react-bootstrap-modal';
+import 'react-bootstrap-modal/lib/css/rbm-patch.css';
+import PropTypes from 'prop-types';
+
+class DownloadDialog extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpen: this.props.isOpen,
+      width: 3000,
+      height: 3000,
+      fileName: '',
+      fileType: 'png',
+      showAnnotations: true,
+      fileTypeOptions: [
+        {
+          key: 'jpg',
+          value: 'jpg'
+        },
+        {
+          key: 'png',
+          value: 'png'
+        }
+      ],
+      previewElementRef: null
```

Comment:
Out of curiosity, why do we track this?

Reply:
This 'eccentric' way to pass trough a ref to state is needed now that
   * 'static' method getDerivedStateFromProps doesnt have access to the instance(this),
   * different of the old way with componentWillReceiveProps
   * See https://fb.me/react-async-component-lifecycle-hooks for details.